shared/src/store/action-reducers/exercise.ts

```diff
@@ -80,11 +79,26 @@ export namespace ExerciseActionReducers {
             // Refresh patient status
             patientUpdates.forEach((patientUpdate) => {
                 const currentPatient = draftState.patients[patientUpdate.id]!;
-                currentPatient.currentHealthStateId = patientUpdate.nextStateId;
-                currentPatient.health = patientUpdate.nextHealthPoints;
+                currentPatient.currentHealthStateName =
+                    patientUpdate.nextStateName;
                 currentPatient.stateTime = patientUpdate.nextStateTime;
                 currentPatient.treatmentTime = patientUpdate.treatmentTime;
-                currentPatient.realStatus = getStatus(currentPatient.health);
+                if (currentPatient.treatmentHistory.length === 0) {
+                    for (let i = 0; i <= (60 * 1000) / tickInterval; i++) {
+                        currentPatient.treatmentHistory.push({
+                            gf: 0,
+                            material: 0,
+                            notarzt: 0,
+                            notSan: 0,
+                            rettSan: 0,
+                            san: 0,
+                        });
+                    }
+                }
+                currentPatient.treatmentHistory.shift();
+                currentPatient.treatmentHistory.push(
+                    patientUpdate.newTreatment
+                );
```

Comment:
This is probably a big performance problem as this means that for every tick for every patient, a new array with 60 items is created (immutability via the immer.js proxy).

Here is a proposal for a better solution:

This value probably doesn't change very often. Therefore, one could compress intervals with the same treatment status.

```ts
interface Interval {
    /**
     * Corresponds to the exerciseTime
     */
    startTime: number,
    value: { gf: number, .... }
}

class TreatmentHistory {
    // in seconds
    static readonly maxLength = 60;

    // If the newTreatment is different to the treatment in the last interval, add it (with the current exerciseTime) 
    // and remove all intervals that are older than (exerciseTime - maxLength)
    // else do nothing
    static addTreatment(intervalls: Intervall[], newTreatment: {gf: number, ...}, exerciseTime)

    static getAverageTreatment(...)
}
```

This would reduce write time (expensive due to immutability) and slightly increase read time (not done very often).

One could also either remove the PatientUpdates from the Tick action or at least only send those for patients that changed their treatment. This would reduce the state-history size by a lot and could either slightly reduce or increase the performance (more calculation on the client vs. less parsing, reduced usage of the WebSocket connection, and less overhead on the server). This would also have big repercussions on the migrations.


Guidelines: "Premature optimization is the root of all evil.", Use the benchmark script.